contrib/mesos/pkg/executor/executor.go

```diff
@@ -1044,3 +1002,220 @@ func executorInfoToAnnotations(ei *mesos.ExecutorInfo) (annotations map[string]s
 
 	return
 }
+
+// implements txTaskOperator, uses the executor state lock to handle concurrent invocation
+func (k *Executor) doForTask(taskId string, f func(*kuberTask, *api.Pod) bool) (changed bool, err error) {
+	k.lock.Lock()
+	defer k.lock.Unlock()
+
+	// exclude tasks which are already deleted from our task registry
+	task := k.tasks[taskId]
```

Comment:
registry is definitely overkill for this PR. i'd like tackle that as part of a larger redesign of pod update processing in the executor as a whole.